doc/vxlan/data_plane_VXLAN_HLD.md

```diff
@@ -0,0 +1,510 @@
+* Per each remote VTEP - tunnel start and termination created.
+
+![Devices connection](images/basic_approach_device_overview.png)
+
+Local traffic will be mapped accoding to VLAN-VNI configuration and send to port which represent correspong remote VTEP.
+
+![Tunnel start traffic](images/basic_approach_tunnel_start.png)
+
+Traffic received on ports, that represent remote VTEPs, will be tunnel terminated, remote MAC will be learned on this remote port and traffic will be mapped according to VLAN-VNI map to local port.
+
+![Tunnel termination traffic](images/basic_approach_tunnel_term.png)
+
+
+### Architecture Design
+
+No new blocks are required.
+
+
+#### Container
+
+No new container is added. The changes are added to existing container SwSS. The details of the changes will be discussed in the Design section below.
+
+
+
+---
+##### Figure 1. Current architecture
+![Current VXLAN ARchitecture](images/curent_vxlan_arch.png)
+---
+
+##### Figure 1. Dataplane architecture
+![Dataplane VXLAN ARchitecture](images\dt_vxlan_arch.png)
+---
+
+#### SAI API
+No changes to SAI APIs are required.
+
+### 2. Configuration and management
+
+#### CLI Enhancements
+
+##### 5.1 Remote VTEP IP configuration
+
+* Ability to add and remove remote VTEP
+   - config vxlan remote add <vtepname> <ipv4-address>
+   - config vxlan remote del <vtepname>
+   - vtepname is a string. 
+   - ipv4-address is an IPV4 address in dotted notation A.B.C.D
+
+##### 5.2 Remote VTEP-VNI association configuration 
+
+* Ability to add and remove remote VTEP assiciation to VNI
+   - config vxlan remote map add <vtepname> <vni-id>
+   - config vxlan remote map del <vtepname> <vni-id>
+   - vtepname is a string. 
+   - ipv4-address is an IPV4 address in dotted notation A.B.C.D
+
+#### Yang Enhancements
+
+##### VXLAN_REMOTE_VTEP
+
+container VXLAN_REMOTE_VTEP {
+
+    description "config db VXLAN_REMOTE_VTEP table";
+
+    list VXLAN_REMOTE_VTEP_LIST {
+
+        key "name";
+        max-elements 1;
+
+        leaf name {
+        /* vni devices are created of the form 'name'-vlanid
+            The kernel has a max limit of 15 chars for netdevices.
+            keeping aside 5 chars for hyphen and vlanid the
+            name should have a max of 10 chars */
+
+            type string {
+                length 1..10;
+            }
+        }
+
+        leaf src_ip {
+            type inet:ip-address;
+        }
+    }
+}
+
+##### VXLAN_REMOTE_VTEP_VNI
+
+container VXLAN_REMOTE_VTEP_VNI {
+
+    description "config db VXLAN_REMOTE_VTEP_VNI table";
+
+    list VXLAN_REMOTE_VTEP_VNI_LIST {
+        key "name mapname";
+
+        leaf name {
+            type leafref {
+                path "/svxlan:sonic-vxlan/svxlan:VXLAN_REMOTE_VTEP_LIST/svxlan:VXLAN_REMOTE_VTEP_VNI_LIST/svxlan:name";
+            }
+        }
+
+        leaf mapname {
+            type string;
+        }
+
+        leaf vni {
+            mandatory true;
+            type stypes:vnid_type;
+        }
```

Comment:
this mean per leaf and map one remove vtep  ?

Reply:
Yes, it is either this or lsit of vni's per remote VTEP